stix2validator/output.py

```diff
@@ -140,30 +140,26 @@ def print_results(results):
     """Prints `results` (the results of validation) to stdout.
 
     Args:
-        results: A dictionary of ValidationResults instances. The key is the
-            file path to the validated document.
+        results: A list of FileValidationResults instances.
 
     """
-    if not hasattr(results, 'items'):
-        results_arr = {'': results}
-        results = results_arr
-
-    level = 0
-    for fn, result in sorted(iteritems(results)):
+    for file_result in results:
         print_horizontal_rule()
-        print_level("[-] Results for: %s", level, fn)
+        print_level("[-] Results for: %s", 0, file_result.filepath)
 
-        if result.is_valid:
+        if file_result.is_valid:
             marker = _GREEN + "[+]"
             verdict = "Valid"
         else:
             marker = _RED + "[X]"
             verdict = "Invalid"
-        print_level("%s STIX JSON: %s", level, marker, verdict)
-
-        if result.warnings:
-            print_warning_results(result, level)
-        if result.errors:
-            print_schema_results(result, level)
-        if result.fatal:
-            print_fatal_results(result.fatal, level)
+        print_level("%s STIX JSON: %s", 0, marker, verdict)
+
+        for object_result in file_result.object_results:
+            if object_result.warnings:
+                print_warning_results(object_result, 1)
+            if object_result.errors:
+                print_schema_results(object_result, 1)
+
+        if file_result.fatal:
+            print_fatal_results(file_result.fatal, 1)
```

Comment:
The output had to change to something more hierarchical: each file can have multiple results.  So I changed this to a two-level indented format.  The first level lists the files, and the second level lists the errors/warnings in each file.

Reply:
@clenk, should that be addressed in this PR or separately?

Reply:
It should be easy enough to add in this PR; just pass `level` to `print_level` instead of `level + 1` in those two functions.